python/nano/src/bigdl/nano/pytorch/inference/multi_instance.py

```diff
@@ -24,40 +24,42 @@
 
 
 class _MultiInstanceModel(torch.nn.Module):
-    def __init__(self, model, ps, send_queues, recv_queues):
+    def __init__(self, model, ps, send_queue, recv_queue):
         super().__init__()
         self.model = model
         self.ps = ps
         self.p_num = len(ps)
-        self.send_queues = send_queues
-        self.recv_queues = recv_queues
+        self.send_queue = send_queue
+        self.recv_queue = recv_queue
 
     def forward(self, input_data: Union[DataLoader, List]) -> List:
         if isinstance(input_data, (DataLoader, list)):
             length = len(input_data)
         else:
             invalidInputError(False, "The input should be a DataLoader or a list of input batchs")
 
-        for idx, batch in enumerate(input_data):
-            self.send_queues[idx % self.p_num].put(batch)
+        for idx_and_batch in enumerate(input_data):
+            self.send_queue.put(idx_and_batch)
 
         outputs = []
-        for idx in range(length):
-            output = self.recv_queues[idx % self.p_num].get()
-            invalidOperationError(not isinstance(output, Exception), f"{output}")
-            outputs.append(output)
-        return outputs
+        for _i in range(length):
+            idx_and_output = self.recv_queue.get()
+            invalidOperationError(not isinstance(idx_and_output, Exception), f"{idx_and_output}")
+            outputs.append(idx_and_output)
+
+        outputs = sorted(outputs, key=lambda x: x[0])
```

Comment:
do you need to sort here? You may store the result using an array directly based on the index, yes?

Reply:
changed it in #6893